### DISCUSSION TITLE
lazyconstraint

### DISCUSSION THREAD

**Discussion by BYS543** [yalmip/YALMIP#1007] [Optimization modelling] [answered]
Hi ,Johan
How to use the attribute ‘lazyconstraint’ in Yalmip. I find the attribute is 0(gurobi). I want to add some constraints before solving.And Is there any difference between adding constraints directly to the model and using the lazyconstraint attribute？

**Comment by johanlofberg** [comment, marked as answer]
It's obsoleted and to be removed

**Reply by johanlofberg** [reply]
...unless you are talking about the Gurobi options LazyConstraints, but then it is simply not relevant as callback structures aren't supported in the MATLAB Gurobi API

**Reply by BYS543** [reply]
Thank you, Johan. `It's obsoleted and to be removed.` This means using `LazyConstraints` to add constraints before solving is the same as starting by putting constraints directly into the model?

**Reply by johanlofberg** [reply]
I don't even know which LazyConstraints operator you are talking about. There was something tested in yalmip some 15 years ago or so, but I have no idea where it is, if it is still there, or what it did really

**Reply by BYS543** [reply]
It is `ops.gurobi.LazyConstraints`

**Reply by johanlofberg** [reply]
That options is irrelevant in MATLAB as MATLAB does not support callbacks, hence you would never be able to return constraints on the fly (and even if gurobi did support callbacks, YALMIP does not support any kind of lazy constraints or callback logic)

**Reply by BYS543** [reply]
got it,thank you.

**Reply by BYS543** [reply]
By the way, is it possible of Matlab to have callback function,maybe future?

**Reply by johanlofberg** [reply]
That is something you would have to ask Gurobi about

**Reply by BYS543** [reply]
So this is business problem,not technology problem?

**Reply by johanlofberg** [reply]
Of course, there is nothing hard about supplying callback functionality, I guess the demand is not large enough for the effort
